core/src/main/java/org/jdbi/v3/core/argument/internal/PojoPropertyArguments.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jdbi.v3.core.argument.internal;
+
+import java.util.Optional;
+
+import org.jdbi.v3.core.argument.NamedArgumentFinder;
+import org.jdbi.v3.core.mapper.reflect.internal.PojoProperties;
+import org.jdbi.v3.core.mapper.reflect.internal.PojoProperties.PojoProperty;
+import org.jdbi.v3.core.mapper.reflect.internal.PojoTypes;
+import org.jdbi.v3.core.statement.StatementContext;
+import org.jdbi.v3.core.statement.UnableToCreateStatementException;
+
+/**
+ * This class hosts the logic from BeanPropertyArguments.
+ * When we can remove that class from public API, this class will easily replace it.
+ */
+public class PojoPropertyArguments extends MethodReturnValueNamedArgumentFinder {
+    private final PojoProperties<?> properties;
+    private final StatementContext ctx;
+
+    public PojoPropertyArguments(String prefix, Object obj, StatementContext ctx) {
+        this(prefix,
+                obj,
+                ctx.getConfig(PojoTypes.class).propertiesOf(obj.getClass())
```

Comment:
Can we name this method `findFor()` to stay consistent with other registries?

Reply:
I think that makes sense here, the method signature clarifies "find _what_ for this" enough IMO.